### PR TITLE
Nevin/fix download telemetry

### LINKED ISSUE #5213: [Telemetry] Download spec review
data spec: https://docs.google.com/spreadsheets/d/1o8p47imE-MkaArVYqRxIR5Eh2ZQzlC3vfdUuIgfcONc/edit#gid=888392947

Spec: https://docs.google.com/document/d/1Qjtt4Zatzb5CWV2qDBob-rftswhXmg4WizbcHmqPr30/edit#

### PR DESCRIPTION
**Pull request by cnevinc**
closes #5213

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/download/GetDownloadFileHeaderTask.java [hunk 1]
```diff
@@ -32,16 +36,22 @@ protected HeaderInfo doInBackground(String... params) {
...
+                headerInfo.contentLength = Long.parseLong(strContentLength);
+            }
             connection.getResponseCode();
             connection.disconnect();
 
         } catch (SSLHandshakeException e) {
             headerInfo.isValidSSL = false;
         } catch (IOException e) {
-            e.printStackTrace();
+            Log.e(TAG, "IOException");
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/focus/download/GetDownloadFileHeaderTask.java:52]: Why not keep `e.printStackTrace()` to show detail call stack for debugging?

File: app/src/main/java/org/mozilla/rocket/download/DownloadInfoRepository.kt [hunk 2]
```diff
@@ -80,8 +80,12 @@ class DownloadInfoRepository {
 
     fun deleteFromDownloadManager(downloadId: Long) {
 
+        DownloadInfoManager.getInstance().downloadManager.remove(downloadId)
+    }
+
+    fun trackDownloadCancel(downloadId: Long): Boolean {
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/rocket/download/DownloadInfoRepository.kt:86]: The returned boolean value is not used. Do we still need to have return value?

File: app/src/main/java/org/mozilla/focus/download/GetDownloadFileHeaderTask.java (hunk 1 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/download/GetDownloadFileHeaderTask.java:53]: fixed

File: app/src/main/java/org/mozilla/focus/download/EnqueueDownloadTask.java [hunk 3]
```diff
@@ -107,7 +107,9 @@ protected ErrorCode doInBackground(Void... params) {
                     public void onInsertComplete(long id) {
                         try {
                             GetDownloadFileHeaderTask.HeaderInfo headerInfo = new GetDownloadFileHeaderTask().execute(download.getUrl()).get();
-                            TelemetryWrapper.startDownloadFile(downloadInfo.getDownloadId().toString(), download.getContentLength(), headerInfo.isValidSSL, headerInfo.isSupportRange);
+                            long contentLengthFromDownloadRequest = download.getContentLength();    // it'll be -1 if it's from context menu, and real file size from webview callback
```
[the remark this answers is not in the document]
Comment (cnevinc) [reply app/src/main/java/org/mozilla/focus/download/EnqueueDownloadTask.java:110]: fixed

File: app/src/main/java/org/mozilla/rocket/download/DownloadInfoRepository.kt (hunk 2 above)
Comment (cnevinc) [reply app/src/main/java/org/mozilla/rocket/download/DownloadInfoRepository.kt]: fixed

**Pull request merged**